sdk/core/Azure.Core/src/DynamicData/MutableJsonChange.cs

```diff
@@ -62,13 +63,41 @@ internal JsonElement GetSerializedValue()
 
         internal bool IsDescendant(string path)
         {
-            if (path.Length > 0)
+            return IsDescendant(path.AsSpan());
+        }
+
+        internal bool IsDescendant(ReadOnlySpan<char> ancestorPath)
+        {
+            return IsDescendant(ancestorPath, Path.AsSpan());
+        }
+
+        internal static bool IsDescendant(ReadOnlySpan<char> ancestorPath, ReadOnlySpan<char> descendantPath)
+        {
+            if (ancestorPath.Length == 0)
             {
-                // Restrict matches (e.g. so we don't think 'a' is a parent of 'abc').
-                path += MutableJsonDocument.ChangeTracker.Delimiter;
+                return descendantPath.Length > 0;
             }
 
-            return Path.StartsWith(path, StringComparison.Ordinal);
+            // Restrict matches (e.g. so we don't think 'a' is a parent of 'abc').
+            char[]? tempArray = null;
+            int length = ancestorPath.Length + 1;
+            Span<char> pathWithDelimiter = length <= MutableJsonElement.MaxStackLimit ?
+                stackalloc char[length] :
+                tempArray = ArrayPool<char>.Shared.Rent(length);
+
+            try
+            {
+                ancestorPath.CopyTo(pathWithDelimiter);
+                pathWithDelimiter[ancestorPath.Length] = MutableJsonDocument.ChangeTracker.Delimiter;
+                return descendantPath.StartsWith(pathWithDelimiter, StringComparison.Ordinal);
```

Comment:
I am not sure I understand why you are doing all the resizing, copying, etc. instead of:
```return descendantPath.Length > ancestorPath.Length && descendantPath.StartsWith(ancestorPath) && descendantPath[ancestorPath.Length] == MutableJsonDocument.ChangeTracker.Delimiter);```